plugins/engine-datafusion/src/main/java/org/opensearch/datafusion/search/DatafusionReader.java

```diff
@@ -56,6 +65,23 @@ public DatafusionReader(String directoryPath, CompositeEngine.ReleasableRef<Cata
         System.out.println("File names: " + Arrays.toString(fileNames));
         System.out.println("Directory path: " + directoryPath);
         this.readerHandle = new ReaderHandle(directoryPath, fileNames, this::releaseCatalogSnapshot);
+        this.segmentStatsFuture = new CompletableFuture<>();
+        setupSegmentStatsCompletableFuture(segmentStatsFuture);
```

Comment:
Can we do this lazily? Not all readers may warrant a stats call, and we can possibly avoid the call if some reader view doesn't get a stats call.